tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_controller_component.ts

```diff
@@ -194,4 +194,26 @@ export class LinkedTimeFobControllerComponent {
     // No fobs can pass the lowest step in this graph.
     return 0;
   }
+
+  stepTyped(fob: Fob, step: number) {
+    let newLinkedTime = {...this.linkedTime};
+    if (fob === Fob.START) {
+      newLinkedTime.start = {step};
+    } else if (fob === Fob.END) {
+      newLinkedTime.end = {step};
+    }
+
+    if (
+      newLinkedTime.end !== null &&
```

Comment:
By the way, I was wrong about claiming newLinkedTime.end could be 0. I forgot that it's actually newLinkedTime.end.step that could be 0. My bad.

You could change this back to !newLinkedTime.end and you definitely wouldn't need any tests for the case where newLinkedTime.end is 0 (because that would be a compile error?)